pymatgen/core/structure.py

```diff
@@ -177,12 +177,12 @@ def __getitem__(self, idx: int | slice):
         """Make neighbor Tuple-like to retain backwards compatibility."""
         return (self, self.nn_distance, self.index, self.image)[idx]
 
-    def as_dict(self) -> dict:  # type: ignore
+    def as_dict(self) -> dict:  # type: ignore[override]
```

Comment:
I guess use of "too general" `# type: ignore` should be avoided wherever possible.